### ISSUE TITLE
WebAssembly hot reload reverts to old version of a type

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/runtime#50190]
### Description

It's possible this is a known limitation and that @lambdageek has it on a backlog, but @pranavkm said I should report things to ensure they are tracked.

The WebAssembly hot reload support is amazingly fast (*really* amazingly - I've never seen hot reload so fast on any other tech) but currently it appears to be limited to making changes in a single type. If you have type `A` referencing `B`, and you edit `B` then the change appears, but if you then edit `A`, your change to `A` appears but it reverts to referencing the old unedited version of `B`.

More precise steps:

 * Use latest preview 3 SDK
 * `dotnet new blazorwasm -o MyApp`
 * `cd MyApp`

Issue 1:

 * Edit `Pages/Index.html`, adding `<Counter />`
 * `dotnet watch`. See your counter on the home screen.
 * Edit `Pages/Counter.razor`, e.g., changing the `<h1>` element to `<h1>Counter2</h1>`. See this show up on the home screen where you added it.
 * Edit `Pages/Index.razor` to add a second `<Counter />`.
    * Expected: New counter appears also with title `Counter2`
    * Actual: New counter appears, but the title is `Counter`, and the other counter component also reverts its title to `Counter`. 

Issue 2:

 * `dotnet watch`
 * Edit `Shared/SurveyPrompt.razor`, changing some text (e.g., `tell us what you think` -> `HELLO`). See your change appear correctly in the browser window.
 * Go to `Index.razor` and try to edit the `<h1>` text.
   * Expected: change shows up
   * Actual: Nothing happens

### Configuration

This is with SDK `6.0.100-preview.3.21173.6`, with `Microsoft.NETCoreSdk.BundledVersions.props` patched to use the runtime version `6.0.0-preview.4.21174.4`.

**Comment by lambdageek** [comment]
Can reproduce the issue (both on wasm and console).  Investigating why we're picking up the wrong method body

**Comment by lambdageek** [comment]
The problem is in `get_method_update_rva` in the case where there are multiple generations, if the lookup in a later generation returns NULL because that generation didn't update the method definition, we treat that as a valid 0 value and return to the caller who interprets it as "no updates at all" and returns the base method.

tl;dr I fail at writing a lookup function. 😓

**Comment by lambdageek** [comment]
Reopening until the preview 3 backport is merged

**Comment by lambdageek** [comment]
Fixed in preview 3 now
